sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -208,14 +237,121 @@ _rom_ext_start_boot:
    */
   la gp, __global_pointer$
 
-  // Re-enable linker relaxation.
-  .option pop
-
   /**
-   * Jump to C Code
+   * Jump to C Code using `jr` to make sure we are jumping to the virtual address.
    */
   .extern rom_ext_main
-  tail rom_ext_main
+  la t0, rom_ext_main
+  jr t0
 
   // Set size so this function can be disassembled.
   .size _rom_ext_start_boot, .-_rom_ext_start_boot
+
+
+/**
+ * Configure the ibex simple address translation 
+ *
+ * @param a0 virtual address
+ * @param a1 size of the region
+ * @param a2 phisical address 
+ */
+  .globl _ibex_address_translation_set
+  .type _ibex_address_translation_set, @function
+_ibex_address_translation_set:
```

Comment:
It might be better to avoid using functions and do this work inline. Using calls means we need to worry about whether calls are PIC or not.